### PR TITLE
Customize tls setup

### PR DESCRIPTION
**Pull request by dzhengfy**
Before this PR, we always scp all required files like cakey, cacert,
severkey ... to the TLS server host and client host which are normal
way for TLS connection setup. But sometimes we need partial setup.
So this PR is to enable the capability to allow the user to define
which files will be scped to the TLS server/client host.

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
@Yingshun Please help review.

File: virttest/utils_conn.py [hunk 1]
```diff
@@ -1292,17 +1300,18 @@ def server_setup(self, on_local=False):
...
-                        serverkey_path: self.libvirt_pki_private_dir}
             cmd = "mkdir -p {0}".format(self.pki_CA_dir)
             status, output = server_session.cmd_status_output(cmd)
             if status:
                 raise ConnMkdirError(self.pki_CA_dir, output)
+            params_scp = {'cacert': cacert_path,
+                          'cakey': cakey_path,
+                          'serverkey': serverkey_path,
+                          'servercert': servercert_path}
+        scp_dict = self._update_scp_list(self.scp_list_server, params_scp)
```
Comment (chunfuwen) [inline virttest/utils_conn.py:1314]: I notice that previous implementation don't this step. Is the step conditionally required or necessary required now?

File: virttest/utils_conn.py [hunk 2]
```diff
@@ -1458,6 +1467,42 @@ def server_setup(self, on_local=False):
                                "%s %s" % (self.client_ip, hostname)}
             self.server_hosts.sub_else_add(pattern_to_repl)
 
+    def _update_scp_list(self, scp_list, params):
+        """
+        Generate the scp file list to remote host
+
+        :param params: dict, the required parameters are like:
```
Comment (chunfuwen) [inline virttest/utils_conn.py:1474]: mismatch doc string order

**Comment by dzhengfy** [comment]
@Yingshun please help have a look. thanks.

File: virttest/utils_conn.py [hunk 3]
```diff
@@ -1292,17 +1300,18 @@ def server_setup(self, on_local=False):
             raise ConnMkdirError(self.libvirt_pki_private_dir, output)
 
         if scp_new_cacert == 'no':
-            scp_dict = {servercert_path: self.libvirt_pki_dir,
-                        serverkey_path: self.libvirt_pki_private_dir}
+            params_scp = {'serverkey': serverkey_path,
+                          'servercert': servercert_path}
```
Comment (Yingshun) [inline virttest/utils_conn.py:1304]: Is scp_new_cecert still useful? can we set to scp_list_xxx directly?

Comment (dzhengfy) [reply virttest/utils_conn.py:1304]: scp_new_cecert is the indicator to decide what files will be scped. So I think it is still useful.

File: virttest/utils_conn.py (hunk 1 above)
Comment (dzhengfy) [reply virttest/utils_conn.py:1314]: If you are asking to invoke for _update_scp_list(), then this step is necessary for all. It is a new created function which applies to all. If the user does not provide scp_list_client/scp_list_server, then invoking  this function makes no effect on them. It is just unified entry.

File: virttest/utils_conn.py (hunk 2 above)
Comment (dzhengfy) [reply virttest/utils_conn.py:1475]: updated, thanks

File: virttest/utils_conn.py [hunk 4]
```diff
@@ -937,7 +938,8 @@ class TLSConnection(ConnectionBase):
                  'server_saslconf', 'server_qemuconf', 'client_qemuconf',
                  'server_libvirtd_tls_socket', 'client_libvirtd_tls_socket',
                  'special_cn', 'server_setup_local', 'server_info_ip',
-                 'daemon_conf', 'daemon_socket_conf')
+                 'daemon_conf', 'daemon_socket_conf', 'scp_list_server',
```
Comment (chunfuwen) [inline virttest/utils_conn.py:941]: please add those parameters description after line 928

File: virttest/utils_conn.py [hunk 5]
```diff
@@ -1292,17 +1300,18 @@ def server_setup(self, on_local=False):
...
         else:
-            scp_dict = {cacert_path: self.pki_CA_dir,
-                        cakey_path: self.pki_CA_dir,
-                        servercert_path: self.libvirt_pki_dir,
-                        serverkey_path: self.libvirt_pki_private_dir}
             cmd = "mkdir -p {0}".format(self.pki_CA_dir)
             status, output = server_session.cmd_status_output(cmd)
             if status:
                 raise ConnMkdirError(self.pki_CA_dir, output)
+            params_scp = {'cacert': cacert_path,
```
Comment (chunfuwen) [inline virttest/utils_conn.py:1310]: Here what params_scp makes difference with params_scp in line 1303  are newly added two keys:cacert,cakey. 
So the proper way could be using params_scp.update(xx)

File: virttest/utils_conn.py [hunk 6]
```diff
@@ -1458,6 +1467,42 @@ def server_setup(self, on_local=False):
                                "%s %s" % (self.client_ip, hostname)}
             self.server_hosts.sub_else_add(pattern_to_repl)
 
+    def _update_scp_list(self, scp_list, params):
+        """
+        Generate the scp file list to remote host
```
Comment (chunfuwen) [inline virttest/utils_conn.py:1472]: Generate -- > update ?

**Review by chunfuwen — COMMENTED** [review]
see inline comments

File: virttest/utils_conn.py (hunk 4 above)
Comment (dzhengfy) [reply virttest/utils_conn.py:941]: update

File: virttest/utils_conn.py (hunk 5 above)
Comment (dzhengfy) [reply virttest/utils_conn.py:1310]: This is just to define a new dict variable (params_scp) with different initial values. params_scp is not defined before this statement. So I think params_scp.update() is not proper.

File: virttest/utils_conn.py (hunk 6 above)
Comment (dzhengfy) [reply virttest/utils_conn.py:1472]: Update

**Review by chunfuwen — APPROVED** [review]
LGTM

**Review by Yingshun — APPROVED** [review]

**Review by kamilvarga — APPROVED** [review]
LGTM.

**Pull request merged**